oracleservice/report_parameters_reader.py

```diff
@@ -55,78 +54,113 @@ def get_stash_staking_parameters(self, stash: Keypair, block_hash: str) -> dict:
-            module='System',
-            storage_function='Account',
-            params=[stash.ss58_address],
-            block_hash=block_hash,
-        )
+        try:
+            account_info = self.service_params.substrate.query(
+                module='System',
+                storage_function='Account',
+                params=[stash.ss58_address],
+                block_hash=block_hash,
+            )
+        except EXPECTED_NETWORK_EXCEPTIONS as exc:
+            logger.warning(f"Failed to get the account {stash.ss58_address} info: {exc}")
+            raise exc
+        except Exception as exc:
+            logger.error(f"Failed to get the account {stash.ss58_address} info: {exc}")
+            raise exc
         metrics_exporter.total_stashes_free_balance.inc(account_info.value['data']['free'])
 
         return account_info.value['data']['free']
 
-    def _get_stake_status(self, stash: Keypair, block_hash: str = None) -> int:
+    def _get_stake_status(self, stash: Keypair, block_hash: str) -> int:
         """
         Get stash account status.
         0 - Idle, 1 - Nominator, 2 - Validator, 3 - None
         """
-        if block_hash is None:
-            block_hash = self.service_params.substrate.get_chain_head()
-
-        staking_validators = self.service_params.substrate.query(
-            module='Session',
-            storage_function='Validators',
-            block_hash=block_hash,
-        )
-
-        staking_nominators = self.service_params.substrate.query_map(
-            module='Staking',
-            storage_function='Nominators',
-            block_hash=block_hash,
-        )
+        try:
+            staking_nominators = self.service_params.substrate.query_map(
+                module='Staking',
+                storage_function='Nominators',
+                block_hash=block_hash,
+            )
+        except EXPECTED_NETWORK_EXCEPTIONS as exc:
+            logger.warning(f"Failed to get nominators: {exc}")
+            raise exc
+        except Exception as exc:
+            logger.error(f"Failed to get nominators: {exc}")
+            raise exc
 
         nominators = set(nominator.value for nominator, _ in staking_nominators)
-        validators = set(validator for validator in staking_validators.value)
-
         if stash.ss58_address in nominators:
             return 1
 
+        try:
+            staking_validators = self.service_params.substrate.query(
+                module='Session',
+                storage_function='Validators',
+                block_hash=block_hash,
+            )
+        except EXPECTED_NETWORK_EXCEPTIONS as exc:
+            logger.warning(f"Failed to get validators: {exc}")
+            raise exc
+        except Exception as exc:
+            logger.error(f"Failed to get validators: {exc}")
+            raise exc
+
+        validators = set(validator for validator in staking_validators.value)
         if stash.ss58_address in validators:
             return 2
```

Comment:
`0` or `3` are never returned. If address is neither nominator or validator, the function will return `None` instead of `0` or `3`. I don't see how it breaks the code right now, but we might get an error if in the future we'll use this method in a way which expects it to return an integer, but not `None`

Reply:
> `0` or `3` are never returned. If address is neither nominator or validator, the function will return `None` instead of `0` or `3`. I don't see how it breaks the code right now, but we might get an error if in the future we'll use this method in a way which expects it to return an integer, but not `None`

`3` is never returned, you are right - that is a deprecated function description. But `0` is returned if an address is neither nominator or validator (look at the line 164). I changed the ordering to decrease an amount of requests to RPC. Those are lazy queries that are executed when accessing a variable.

Reply:
Yep, last line with `return 0` wasn't displayed for me in changes list